### ISSUE TITLE
Moq1100 incorrectly firing on nullable parameters

### ISSUE THREAD

**Issue by marcovr** [rjmurillo/moq.analyzers#172]
Analyzer Moq1100 is incorrectly firing in the following case:
```cs
public interface IFoo
{
    bool DoSomething(object? bar);
}

var mock = new Mock<IFoo>();
mock.Setup(m => m.DoSomething(It.IsAny<object?>())).Returns((object? bar) => true);
```

The analyzer reports:
> Moq1100: Callback signature must match the signature of the mocked method

Note: this not only happens with `object` but also other types

**Comment by rjmurillo** [comment]
@marcovr Thank you for reporting this issue. Would you be able to try a private to ensure this issue is corrected? [Moq.Analyzers.0.2.0-g7dd22517f7.nupkg](https://github.com/rjmurillo/moq.analyzers/actions/runs/10566077819/artifacts/1856241340)

**Comment by marcovr** [comment]
Thanks @rjmurillo!
I can confirm that the preview package resolves the issue 👍🏻.

However, I noticed that it introduced another (possibly unintended?) change in behaviour. Consider the following example:
```cs
var mock = new Mock<IFoo>();
mock.Setup(m => m.DoSomething(42)).Returns((long bar) => true);

public interface IFoo
{
    bool DoSomething(long bar);
}
```

In this case, the analyzer now reports `Moq1100` unless I either perform a cast:
```cs
mock.Setup(m => m.DoSomething((long)42)).Returns((long bar) => true);
```
or change the signature of the return lambda:
```cs
mock.Setup(m => m.DoSomething(42)).Returns((int bar) => true);
```

I don't really like the variant with a cast as it causes other analyzers to report an unnecessary cast. And the other variant works fine, but seems kind of backwards to me. Because I now need a "wrong" signature to make the analyzer happy, which checks for correct signatures 😁

**Comment by rjmurillo** [comment]
Thanks @marcovr for your patience and the additional test cases; I've added them to the suite.

You're right, the updated type conversion detection method caused a regression. In the case there was an `Int32` value being passed to `Int64`, which was failing the now stricter type check. I've updated the implementation since there is a conversion that can happen in a few scenarios:

```csharp
var mock = new Mock<IFoo>();
mock.Setup(m => m.DoSomething(42)).Returns((long bar) => true);
mock.Setup(m => m.DoSomethingElse(42)).Returns((long var) => true);

public interface IFoo
{
    bool DoSomething(long bar);
    bool DoSomethingElse(object bar);
}
```

In the `DoSomethingElse` case, the integers can be boxed to an object, and can be done so on your behalf without an explicit cast. The updated implementation should now permit this without having to write extra code nor conflict with your other analyzers.

I have another private for you to try: [Moq.Analyzers.0.2.0-g78ada5acc1.nupkg](https://github.com/rjmurillo/moq.analyzers/actions/runs/10583789524/artifacts/1861254424)

**Comment by marcovr** [comment]
Thanks for the quick fix @rjmurillo!
I can confirm that it now works as expected and I don't get any false positives anymore 🙂

**Comment by rjmurillo** [comment]
@marcovr Thank you for your help and patience. I've published a new version of the package with this bug fix: https://www.nuget.org/packages/Moq.Analyzers/0.1.2
